build/verify-archive.sh

```diff
@@ -10,8 +10,9 @@ set -euo pipefail
 apt-get update
 apt-get install -y cmake xz-utils
 
-tar xzf cockroach.src.tgz
-(cd cockroach-* && make install)
+workdir="$(mktemp -d)"
```

Comment:
Good point. That reminds me, the current `docker run` invocation in `build/teamcity-verify-archive.sh` doesn't contain `--rm`, which means we're keeping these containers indefinitely. That will definitely have a bad impact on disk usage, so we should fix it before we enable the build step in TC.